From 0ba723034e1df8726c0948a7cfe5ad6f6ef8b62c Mon Sep 17 00:00:00 2001
From: Devarsh Thakkar <devarsht@xilinx.com>
Date: Wed, 7 Mar 2018 23:37:25 -0800
Subject: [PATCH 0165/1851] dmaengine: xilinx_frmbuf: Add support for XV15 and
 XV20 contiguous formats

commit e705c8e46f566d279be0a9126c5e71d6a651a661 from
https://github.com/Xilinx/linux-xlnx.git

This patch updates the framebuffer driver to add support for
XV15 and XV20 contiguous formats.

Signed-off-by: Devarsh Thakkar <devarsht@xilinx.com>
Signed-off-by: Satish Kumar Nagireddy <satishna@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/dma/xilinx/xilinx_frmbuf.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/drivers/dma/xilinx/xilinx_frmbuf.c b/drivers/dma/xilinx/xilinx_frmbuf.c
index 6119a37e3386..4526cbbdd25d 100644
--- a/drivers/dma/xilinx/xilinx_frmbuf.c
+++ b/drivers/dma/xilinx/xilinx_frmbuf.c
@@ -316,6 +316,16 @@ static const struct xilinx_frmbuf_format_desc xilinx_frmbuf_formats[] = {
 		.v4l2_fmt = V4L2_PIX_FMT_XV15M,
 		.fmt_bitmask = BIT(13),
 	},
+	{
+		.dts_name = "xv15",
+		.id = XILINX_FRMBUF_FMT_Y_UV10_420,
+		.bpw = 32,
+		.ppw = 3,
+		.num_planes = 2,
+		.drm_fmt = 0,
+		.v4l2_fmt = V4L2_PIX_FMT_XV15,
+		.fmt_bitmask = BIT(13),
+	},
 	{
 		.dts_name = "xv20",
 		.id = XILINX_FRMBUF_FMT_Y_UV10,
@@ -326,6 +336,16 @@ static const struct xilinx_frmbuf_format_desc xilinx_frmbuf_formats[] = {
 		.v4l2_fmt = V4L2_PIX_FMT_XV20M,
 		.fmt_bitmask = BIT(14),
 	},
+	{
+		.dts_name = "xv20",
+		.id = XILINX_FRMBUF_FMT_Y_UV10,
+		.bpw = 32,
+		.ppw = 3,
+		.num_planes = 2,
+		.drm_fmt = 0,
+		.v4l2_fmt = V4L2_PIX_FMT_XV20,
+		.fmt_bitmask = BIT(14),
+	},
 	{
 		.dts_name = "bgr888",
 		.id = XILINX_FRMBUF_FMT_RGB8,
-- 
2.31.1

